File: 374.cpp
// Forward declaration of guess API.
// @param num, your guess
// @return -1 if my number is lower, 1 if my number is higher, otherwise return 0
int guess(int num);

class Solution {
public:
    int guessNumber(int n) {
        long int low = 0, high = n, mid; 
        
        while (true){
            mid = (low+high)/2; 
            
            if (guess(mid) == 0)
                return mid; 
            else if(guess(mid) == 1)
                low = mid+1; 
            else if (guess(mid) == -1)
                high = mid-1;
        }
    }
};

File: 374.py
class Solution:
    def guessNumber(self, n: int) -> int:
        l = 1 
        while l <= n: 
            mid = (n-l)//2 + l 
            if guess(mid) == 0: 
                return mid 
            elif guess(mid) == 1: 
                l = mid + 1
            elif guess(mid) == -1: 
                n = mid - 1 

        return 1
